pkg/proxy/servicechangetracker_test.go

```diff
@@ -580,7 +580,7 @@ func TestServiceToServiceMap(t *testing.T) {
 					svcInfo.port != expectedInfo.port ||
 					svcInfo.protocol != expectedInfo.protocol ||
 					svcInfo.healthCheckNodePort != expectedInfo.healthCheckNodePort ||
-					!sets.New[string](svcInfo.externalIPs...).Equal(sets.New[string](expectedInfo.externalIPs...)) ||
+					!reflect.DeepEqual(svcInfo.externalIPs, expectedInfo.externalIPs) ||
```

Comment:
In the general case, this is not safe, because there are two ways of representing IPv4 `net.IP`s (the `.To4()` way and the `.To16()` way). So this is not only requiring them to be equal, it's requiring the expected values to have been generated in the same way as the actual values...

Given that we hope to eventually move to `netip.Addr` (which doesn't have this problem), this is probably fine.

Reply:
The problem was that net.IP isn't "comparable", so you can't create a set from them. I know this was a short-cut, but as you say:
> Given that we hope to eventually move to netip.Addr (which doesn't have this problem)

netip.Addr *is* comparable.